File: game-server/src/main/java/com/osroyale/game/event/impl/ItemOnPlayerEvent.java
package com.osroyale.game.event.impl;

import com.osroyale.game.event.Event;
import com.osroyale.game.world.entity.mob.player.Player;
import com.osroyale.game.world.items.Item;

public class ItemOnPlayerEvent implements Event {

    private final Player other;

    private final Item used;

    private final int slot;

    public ItemOnPlayerEvent(Player other, Item used, int slot) {
        this.other = other;
        this.used = used;
        this.slot = slot;
    }

    public Player getOther() {
        return other;
    }

    public Item getUsed() {
        return used;
    }

    public int getSlot() {
        return slot;
    }

}

File: game-server/src/main/kotlin/org/jire/tarnishps/event/player/ItemOnPlayerEvent.kt
package org.jire.tarnishps.event.player

import com.osroyale.game.event.impl.ItemOnPlayerEvent
import com.osroyale.game.plugin.PluginManager
import com.osroyale.game.world.World
import com.osroyale.game.world.entity.mob.player.Player
import com.osroyale.net.packet.out.SendMessage
import org.jire.tarnishps.event.Event
import kotlin.jvm.optionals.getOrNull

/**
 * @author Jire
 */
class ItemOnPlayerEvent(
    val interfaceId: Int,
    val item: Int,
    val itemSlot: Int,
    val slot: Int
) : Event {

    override fun handle(player: Player) {
        val used = player.inventory[itemSlot] ?: return
        if (!used.matchesId(item)) return

        val other = World.getPlayerBySlot(slot).getOrNull() ?: return
        
        player.walkTo(other) {
            player.face(other.position)
            if (!PluginManager.getDataBus().publish(player, ItemOnPlayerEvent(other, used, itemSlot))) {
                player.send(SendMessage("Nothing interesting happens."))
            }
        }
    }

}